src/component/project/ProjectAccess/ProjectAccessAddUser/ProjectAccessAddUser.tsx

```diff
@@ -0,0 +1,241 @@
+import { Search } from '@material-ui/icons';
+import Autocomplete from '@material-ui/lab/Autocomplete';
+import { Alert } from '@material-ui/lab';
+import { ProjectRoleSelect } from '../ProjectRoleSelect/ProjectRoleSelect';
+import useProjectApi from '../../../../hooks/api/actions/useProjectApi/useProjectApi';
+import { useParams } from 'react-router-dom';
+import useToast from '../../../../hooks/useToast';
+import useProjectAccess, {
+    IProjectAccessUsers,
+} from '../../../../hooks/api/getters/useProjectAccess/useProjectAccess';
+import { IProjectRole } from '../../../../interfaces/role';
+import ConditionallyRender from '../../../common/ConditionallyRender';
+
+interface IProps {
+    roles: IProjectRole[];
+}
+
+export const ProjectAccessAddUser = ({ roles }: IProps) => {
+    const { id } = useParams<{ id: string }>();
+    const [user, setUser] = useState<IProjectAccessUsers | undefined>();
+    const [role, setRole] = useState<IProjectRole>({
+        id: -1,
+        name: '',
+        description: '',
+        type: '',
+    });
+    const [options, setOptions] = useState([]);
+    const [loading, setLoading] = useState(false);
+    const { setToastData } = useToast();
+    const { refetchProjectAccess, access } = useProjectAccess(id);
+
+    const { searchProjectUser, addUserToRole } = useProjectApi();
+
+    useEffect(() => {
+        if (roles.length > 0) {
+            const regularRole = roles.find(
+                r => r.name.toLowerCase() === 'regular'
+            );
+            setRole(regularRole || roles[0]);
+        }
+    }, [roles]);
+
+    const search = async (query: string) => {
+        if (query.length > 1) {
+            setLoading(true);
+
+            const result = await searchProjectUser(query);
+            const userSearchResults = await result.json();
+
+            const filteredUsers = userSearchResults.filter(
+                (selectedUser: IProjectAccessUsers) => {
+                    const selected = access.users.find(
+                        (user: IProjectAccessUsers) =>
+                            user.id === selectedUser.id
+                    );
+                    return selected ? false : true;
+                }
+            );
+            setOptions(filteredUsers);
+        } else {
+            setOptions([]);
+        }
+        setLoading(false);
+    };
+
+    const handleQueryUpdate = (evt: { target: { value: string } }) => {
+        const q = evt.target.value;
+        search(q);
+        if (options.length === 1) {
+            return;
+        }
+    };
+
+    const handleBlur = () => {
+        if (options.length > 0) {
+            const user = options[0];
+            setUser(user);
+        }
+    };
+
+    const handleSelectUser = (
+        evt: ChangeEvent<{}>,
+        selectedUser: string | IProjectAccessUsers | null
+    ) => {
+        setOptions([]);
+        if (typeof selectedUser === 'string' || selectedUser === null) {
+            return;
+        }
```

Comment:
Seems like the next `if` would cover these cases as well.

Reply:
Normally. But because of a quirk on how the Material-UI <AutomComplete /> component works you can either these values passed in: `selectedUser: string | IProjectAccessUser | null`

If I remove those checks then ts will complain that "id" doesn't exist on string, and I can't force the AutoComplete field to conform to my handler (at least I haven't found a way).